k256/src/arithmetic/affine.rs

```diff
@@ -171,6 +171,12 @@ impl GroupEncoding for AffinePoint {
         let is_compressed_point = y_is_odd | tag.ct_eq(&sec1::Tag::CompressedEvenY.into());
         Self::decompress(FieldBytes::from_slice(&bytes[1..]), y_is_odd)
             .and_then(|point| CtOption::new(point, is_compressed_point))
+            .or_else(|| {
+                CtOption::new(
+                    AffinePoint::identity(),
+                    tag.ct_eq(&sec1::Tag::Identity.into()),
+                )
+            })
```

Comment:
This seemed to me to be the cleanest solution, but not entirely correct because it doesn't check if the rest of the bytes are also `0`s.

I think the best approach if this is desired is to change `DecompressPoint::decompress()` to handle the identity point, which would require a breaking change.

Reply:
This would accept a `0`-tagged value with garbage bytes. Not sure that's a good idea. It should probably check all of them.

This is a lot of logic to repeat on a per-crate basis. It seems like it might be worth hoisting into the `elliptic-curve` crate in some form. It could probably be composed in terms of `FromDecodedPoint`.

Reply:
I'm unsure what you mean, `FromDecodedPoint::from_encoded_point()` requires an `EncodedPoint` and doesn't do the check we want. Do you mean adding a new method to `FromDecodedPoint`?

Reply:
No, I mean something which uses `FromDecodedPoint` to do the decoding, possibly after changing its result to a `CtOption`, and offloading the SEC1 tag handling there

Reply:
I just realized, I thought you meant `FromEncodedPoint`, I can't find `FromDecodedPoint`.

Reply:
Err yes sorry (responding on my phone here)